File: 2358-number-of-ways-to-split-array/number-of-ways-to-split-array.cpp
class Solution {
public:
    int waysToSplitArray(vector<int>& nums) {
        long long n=nums.size();
        vector<long long>pref(n,0);
        long long ans=0;
        long long sm=0;
        for(int i=0;i<n;i++){
            sm+=nums[i];
            pref[i]=sm;
        }
        for(int i=0;i<n-1;i++){
           
            if(pref[i]>=(pref[n-1]-pref[i+1]+nums[i+1] )){
             //   cout<<i<<endl;
                ans++;
            }
        }
        return ans;
    }
};

File: 2358-number-of-ways-to-split-array/number-of-ways-to-split-array.py
class Solution:
    def waysToSplitArray(self, nums: List[int]) -> int:
        sm=sum(nums)
        cur_sm=0
        ans=0
        for i in range(len(nums)-1):
            cur_sm+=nums[i]
            rig_sm=(sm-cur_sm)
            if cur_sm>=rig_sm:
                ans+=1
        


        return ans